Add mapserver

akrherz opened the pull request:
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

This PR adds Open Source Web Mapping application [Mapserver](https://mapserver.org/).  Mapserver provides a CGI interface used from a web server and a mapscript interface more popular with PHP than python (only python is included here for now :) ).  Some notes on this initial implementation...

1) I struggled with naming this thing and attempting various iterations of sub-packages bundling different parts of mapserver.  There a SO library, some bin/ scripts and the `mapscript` python interface.  I finally settled on a monolith of `mapserver` to hopefully make this most obvious to end-users.
2) I punted on windows for now, I will attempt it once the feedstock is up and running.
3) There's dependency problems still within `conda-forge` and things that touch GDAL.  I have pinned some things down just to get dep solving to work.  I will work to undo this as the bot can help migrate things, etc.  I also continue to actively work to complete the GDAL, GEOS, and proj migrations so to march pinning forward.
4) There's many more options that could be enabled in this build given what conda-forge supports, but I had big time dependency solving troubles, so will need to piecemill it once I can get migrations done.
5) Again with the naming, the python module created is called `mapscript`, but I sort of doubt users would intuitively attempt to `conda install mapscript`, but rather would expect mapserver to be the package name.  If somebody wants to help with splitting things up, I would be happy to work with them.

I am shamelessly pinging @hobu here to see if he has any thoughts on this feedstock creation and if he knows of any others that may be interested in maintaining.  I will ping mapserver-users email list as well after initial feedstock creation.

Thanks for the consideration.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/mapserver```) and found some lint.

Here's what I've got...


For **recipes/mapserver**:

 * ``requirements: host: geos=3.10.1`` must contain a space between the name and the pin, i.e. ``geos =3.10.1``
 * ``requirements: host: libgdal=3.4.0`` must contain a space between the name and the pin, i.e. ``libgdal =3.4.0``
 * ``requirements: host: proj=8.2.0`` must contain a space between the name and the pin, i.e. ``proj =8.2.0``

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/mapserver```) and found it was in an excellent condition.

akrherz commented:
- addressed linting
- added `make` as a requirement, grumble local docker_build never hit that error
- See I have `libiconv` issues on osx, which I had trouble addressing on my testing in Linux.  Will have to punt on that at the moment and intend on addressing asap within the full-blown feedstock.

hobu commented:
I am willing to be a backup maintainer on this if you're going to stick around @akrherz.

hobu commented:
Ideally this wouldn't have the PROJ, GEOS, and GDAL pins, however.

akrherz commented:
> Ideally this wouldn't have the PROJ, GEOS, and GDAL pins, however.

Totally agreed, but alas.  No joy in mudville with my local development resolving deps.  That'll be job one to remove with migrations, etc.

akrherz commented:
Yup, deps did not resolve
```
 - swig
 - giflib 5.2.*
 - geos 3.10.0.*
 - proj 8.0.1.*
 - libpng 1.6.*
 - libxml2 2.9.*
 - postgresql
 - libgdal 3.3.*
 - libgcc-ng >=9.4.0
 - libstdcxx-ng >=9.4.0
 - freetype 2.*
 - jpeg 9.*
 - pip
 - python 3.7.* *_cpython
 - libpq
 - libcurl 7.*

with channels:
 - local
 - conda-forge
 - conda-forge

The reported errors are:
- Encountered problems while solving:
-   - package libgdal-3.3.3-h18e3bf0_2 requires proj >=8.1.1,<8.1.2.0a0,
```

hobu commented:
@ocefpaf Can you help us get a feedstock made for this? I think we're ready to go.

ocefpaf commented on recipes/mapserver/build.sh:
```diff
@@ -0,0 +1,23 @@
+    --debug-output ${SRC_DIR}
+
+make -j${CPU_COUNT}
+make install
```
Do we have a `make check`?

hobu replied:
@akrherz maybe just a print of `mapserv -v` would be sufficient to know we're working.

ocefpaf commented on recipes/mapserver/meta.yaml:
```diff
@@ -0,0 +1,77 @@
+
+build:
+  number: 0
+  # Any volunteers? :)
```
:rofl:

hobu replied:
I was "volunteered" 😄

akrherz replied:
@hobu You volunteered to setup the windows build?  Oh, that is kind of you :)

ocefpaf commented on recipes/mapserver/meta.yaml:
```diff
@@ -0,0 +1,77 @@
+    - python
+    - pip
+    - freetype
+    - geos =3.10.1
```
Is there a reason to not pin these to the same version in run?

akrherz replied:
I thought I hit problems, but will try it shortly, thanks

ocefpaf commented on recipes/mapserver/meta.yaml:
```diff
@@ -0,0 +1,77 @@
+  license_family: MIT
+  license_file: LICENSE.txt
+  summary: 'Platform for publishing spatial data and interative mapping.'
+  # The remaining entries in this section are optional, but recommended.
```
```suggestion
```

ocefpaf reviewed (approved):
Just a few minor comments/questions and this should be ready to go!
